src/pages/explore/api/get-stats.ts

```diff
@@ -0,0 +1,97 @@
+import { ValueView, AssetId } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { pindexer } from '@/shared/database';
+import { DurationWindow } from '@/shared/utils/duration';
+import { toValueView } from '@/shared/utils/value-view';
+
+export interface StatsData {
+  activePairs: number;
+  trades: number;
+  largestPair?: { start: string; end: string };
+  topPriceMover?: { start: string; end: string; percent: number };
+  directVolume: ValueView;
+  liquidity: ValueView;
+  largestPairLiquidity?: ValueView;
+}
+
+export type StatsResponse = StatsData | { error: string };
+
+export const getStats = async (): Promise<StatsResponse> => {
+  try {
+    const durationWindow: DurationWindow = '1d';
+
+    const chainId = process.env['PENUMBRA_CHAIN_ID'];
+    if (!chainId) {
+      return { error: 'PENUMBRA_CHAIN_ID is not set' };
+    }
+
+    const registryClient = new ChainRegistryClient();
+
+    const [registry, results] = await Promise.all([
+      registryClient.remote.get(chainId),
+      pindexer.stats(durationWindow),
+    ]);
+
+    const stats = results[0];
+    if (!stats) {
+      return { error: `No stats found` };
+    }
+
+    // TODO: Add getMetadataBySymbol() helper to registry npm package
+    const allAssets = registry.getAllAssets();
+    // TODO: what asset should be used here?
+    const usdcMetadata = allAssets.find(asset => asset.symbol.toLowerCase() === 'usdc');
+    if (!usdcMetadata) {
+      return { error: 'USDC not found in registry' };
+    }
```

Comment:
question: the database doesn't seem to have the base asset for the global summary. How ok is it to have this check in code?

Reply:
Yeah we can add metadata for this